v2/pkg/protocols/headless/request.go

```diff
@@ -98,14 +98,19 @@ func (request *Request) executeRequestWithPayloads(inputURL string, payloads map
 	gologger.Verbose().Msgf("Sent Headless request to %s", inputURL)
 
 	reqBuilder := &strings.Builder{}
-	if request.options.Options.Debug || request.options.Options.DebugRequests {
+	if request.options.Options.Debug || request.options.Options.DebugRequests || request.options.Options.DebugResponse {
 		gologger.Info().Msgf("[%s] Dumped Headless request for %s", request.options.TemplateID, inputURL)
 
 		for _, act := range request.Steps {
-			reqBuilder.WriteString(act.String())
+			actStepStr := act.String()
+			if strings.Contains(actStepStr, "{{BaseURL}}") {
```

Comment:
We can replace directly, as internally `strings.ReplaceAll` already uses `strings.Index`:
```go
actStepStr = strings.ReplaceAll(actStepStr, "{{BaseURL}}", inputURL)
```